### ISSUE TITLE
Config watcher detects change of secret by I do not see any actual refresh of spring apps

### ISSUE THREAD

**Issue by st-h** [spring-cloud/spring-cloud-kubernetes#1098]
This is for image `springcloud/spring-cloud-kubernetes-configuration-watcher:3.0.0-SNAPSHOT`

In order to be able to deploy the config watcher, I needed to remove the apiGroups `apps` and `extensions`. `extensions` isn't available on our cluster at all. and `apps` only has the following resources: `daemonsets`, `daemonsets`, `replicasets`, `statefulsets`. The apiGroups that were applied successfully are: `""`. The config watcher is running, however I do not see that it triggers refresh on our spring cloud app deployed on the same k8s cluster.

When a secret is refreshed, I see the following log statement:

`o.s.c.k.c.watcher.WatcherUtil            : will schedule remote refresh based on apps : [tests.secret]    `

I also occasionally see the following statements, where I am not sure if they indicate an issue or if they are to be exepected:

```
│ 2022-10-10T14:43:12.275Z  INFO 1 --- [s.V1ConfigMap-1] i.k.c.informer.cache.ReflectorRunnable   : ResourceVersion 793956303 and Watch connection expired: too old resource version: 793956303 (793965024) , will retry w/o resourceVersion next time                  │
│ 2022-10-10T14:43:12.276Z  INFO 1 --- [s.V1ConfigMap-1] i.k.c.informer.cache.ReflectorRunnable   : class io.kubernetes.client.openapi.models.V1ConfigMap#Watch expired, will re-list-watch soon                                                                        │
│ 2022-10-10T14:43:13.276Z  INFO 1 --- [s.V1ConfigMap-1] i.k.c.informer.cache.ReflectorRunnable   : class io.kubernetes.client.openapi.models.V1ConfigMap#Start listing and watching..
```

Long story short: Are the apiGroups `apps` and `extensions` required for the config watcher to auto discover spring applications?

**Comment by wind57** [comment]
There are many thing here, so let's take it each at a time if you do not mind.

- I'll need to refresh my knowledge about `apiGroups` in order to properly answer that, but it seems to me (please correct me if I am wrong) that config-watcher is deployed just fine. And I kind of know it was deployed just fine because of the next point:

- what you see with this message: `o.s.c.k.c.watcher.WatcherUtil : will schedule remote refresh based on apps : [tests.secret]` is an indication that a secret with name `tests.secret` was changed (that's your secret name, right?); we caught this event and will try to schedule a remote refresh based on it. Now, to properly continue this talk, you need to tell us if this is a HTTP or a BUS based config-watcher. If it's http, you can enabled debug logging and when you do that, you will see such a line in logs:

`"Sending refresh request for " + <appName> + " to URI " + <actuatorUri>`

If you enable debug mode, I will get a lot more details and things will be easier to diagnose. 

- those log entries about "resource too old" are absolutely fine. If you want to read more here is a good resource : https://stackoverflow.com/questions/66080942/what-k8s-bookmark-solves/72875470

**Comment by st-h** [comment]
Thank you. Turns out the issue was a missing annotation `spring.cloud.kubernetes.secret.apps: "myapp"` within the secret. Sorry about the noise. However, I have run into another issue where I am unable to configure the actuator path: #1099
